app/services/calculators/multiples/multiple_offenses_calculator.rb

```diff
@@ -36,7 +37,9 @@ def spent_date_for(proceeding)
         #
         convictions_by_date.without(proceeding).each do |conviction|
           other_conviction_date = conviction.conviction_date
-          other_spent_date = conviction.spent_date_without_relevant_orders
+
+          # Pick the spent date from memory if we have it, otherwise get it fresh.
+          other_spent_date = dates_memory.fetch(conviction, conviction.spent_date_without_relevant_orders)
```

Comment:
Having a date in the memory means that the spent date of that conviction has been dragged through by another conviction. So instead of using the spent date of it's own sentences, it uses the date that has been stored in memory.